DOC: Updated the docstring of pandas.Series.str.get

pranavsuri opened the pull request:
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
###################### Docstring (pandas.Series.str.get)  ######################
################################################################################

Extract the element from each component at the specified position.

Extract element from lists, tuples, or strings in each element in the
Series/Index.

Parameters
----------
i : int
    Position of the element to extract.

Returns
-------
items : Series/Index of objects

Examples
--------
>>> s = pd.Series(["String", (1, 2, 3), ["a", "b", "c"], 123])
>>> s
0       String
1    (1, 2, 3)
2    [a, b, c]
3          123
dtype: object

>>> s.str.get(1)
0      t
1      2
2      b
3    NaN
dtype: object

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	See Also section not found

```

pranavsuri committed:
DOC: Updated the docstring of pandas.Series.str.get

datapythonista commented on pandas/core/strings.py:
```diff
@@ -1651,17 +1651,36 @@ def str_translate(arr, table, deletechars=None):
+    1      2
+    2      b
+    3    NaN
+    dtype: object
```
Looks good. May be we could a dictionary to the `Series` too, and an example indexing by a negative number. I'm not sure, as the example could be too complex for the simplicity of the function. But it'd illustrate the different behaviors.

It'd also help to have an explanation before the example, saying something like "When the value is not an iterable, or a dict where the index is not in the keys, `str.get` returns `NaN`".

If you do it you may encounter this bug: #20671. I'd wait until it's fixed, otherwise you'd have to use an index that is a key in the dictionary, and not illustrate the case when the index is not a key, which raises a `KeyError` right now.

pranavsuri replied:
Hello @datapythonista! I see that you've fixed the bug (which I tried fixing, but your solution is a lot better). Should I proceed writing the docstring with a dictionary & a negative value?

pranavsuri commented:
Hello @datapythonista! I see that you've fixed the bug (which I tried fixing, but your solution is a lot better). Should I proceed writing the docstring with a dictionary & a negative value?

datapythonista commented:
You can add it, yes. My changes are not merged, so when you run the validation script you'll get the doctest fail, but no problem with that, when the bug is fixed your doctest will pass and your PR could be merged.

pranavsuri committed:
Updated docstring of pandas.Series.str.get

Added an example with a dictionary, a negative integer and an example with a negative index, i.e., not present in the dictionary.

pep8speaks commented:
Hello @pranavsuri! Thanks for updating the PR.

 - In the file [`pandas/core/strings.py`](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py), following are the PEP8 issues :

> [Line 1670:33](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1670): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 1671:26](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1671): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 1672:32](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1672): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 1673:26](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1673): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 1683:1](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1683): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 1692:1](https://github.com/pandas-dev/pandas/blob/9ce7f42db961c7afafe1e006ac152c8890dadbaf/pandas/core/strings.py#L1692): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

datapythonista commented:
@pranavsuri, Can you fix the PEP-8 issues? Besides that lgtm. But it needs to wait until #20671 is merged.

pranavsuri committed:
Fixed PEP8 issues (trailing whitespaces)

jreback commented:
thanks @pranavsuri
